tls/extensions/s2n_client_key_share.c

```diff
@@ -44,10 +44,60 @@
  * - Key shares for named groups not in the client's supported_groups extension.
  **/
 
-static int s2n_ecdhe_supported_curves_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_client_key_share_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_client_key_share_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_client_key_share_extension = {
+    .iana_value = TLS_EXTENSION_KEY_SHARE,
+    .is_response = false,
+    .send = s2n_client_key_share_send,
+    .recv = s2n_client_key_share_recv,
+    .should_send = s2n_extension_send_if_tls13_connection,
+    .if_missing = s2n_extension_noop_if_missing,
+};
 
-int s2n_extensions_client_key_share_recv(struct s2n_connection *conn, struct s2n_stuffer *extension)
+static int s2n_ecdhe_supported_curves_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+{
+    notnull_check(conn);
+    notnull_check(conn->config);
+
+    const struct s2n_ecc_preferences *ecc_pref = NULL;
+    GUARD(s2n_connection_get_ecc_preferences(conn, &ecc_pref));
+    notnull_check(ecc_pref);
+
+    const struct s2n_ecc_named_curve *named_curve = NULL;
+    struct s2n_ecc_evp_params *ecc_evp_params = NULL;
+
+    for (uint32_t i = 0; i < ecc_pref->count; i++) {
+        ecc_evp_params = &conn->secure.client_ecc_evp_params[i];
+        named_curve = ecc_pref->ecc_curves[i];
```

Comment:
missing not null check

Reply:
ecc_evp_params can't be null, because we just set it to the memory address of ["conn->secure.client_ecc_evp_params[i]"](https://github.com/awslabs/s2n/blob/2a2a7a3cb9d97d1f8730cc6b59ed2b758b327516/tls/s2n_crypto.h#L74). We're assigning it the location of a non-pointer variable, not assigning it the value of a pointer variable.